akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -1634,14 +1651,25 @@ private[akka] class ReadAggregator(
   override val doneWhenRemainingSize = consistency match {
     case ReadFrom(n, _) ⇒ nodes.size - (n - 1)
     case _: ReadAll     ⇒ 0
-    case _: ReadMajority ⇒
+    case ReadMajority(_, minCap) ⇒
       val N = nodes.size + 1
-      val r = N / 2 + 1 // read from at least (N/2+1) nodes
+      val r = calculateReadMajorityWithMinCap(minCap, N)
       N - r
     case ReadLocal ⇒
       throw new IllegalArgumentException("ReadLocal not supported by ReadAggregator")
   }
 
+  // returns minCap or number of nodes if (Nodes/2+1) are less then minCap else (Nodes/2+1)
+  def calculateReadMajorityWithMinCap(minCap: Int, numberOfNodes: Int): Int = {
```

Comment:
I think it would be good to move this to the companion object and add a unit test.